admiral/pkg/controller/common/common.go

```diff
@@ -702,3 +702,24 @@ func GetMeshPortsHelper(meshPorts string, destService *k8sV1.Service, clusterNam
 	}
 	return ports
 }
+
+func GenerateUniqueNameForVS(syncNamespace string, vsName string) string {
+
+	if syncNamespace == "" && vsName == "" {
+		return ""
+	} else if syncNamespace == "" {
+		return vsName
+	} else if vsName == "" {
+		return syncNamespace
+	}
+
+	newVSName := syncNamespace + "-" + vsName
```

Comment:
is this enough to generate a unique name? Wont we have the same issue if two clients share the same name, then we may overwrite them in the sync namespace?

Reply:
+1 maybe lets append a random number?